Opa tests fail in CircleCI and standalone kubeapi server

While this is investigated/fixed, we should exclude the tests to avoid blocking other PRs. 


2015-12-21T01:41:05.191095Z	info	template Kind: tracespan, &InstanceParam{TraceId:,SpanId:,ParentSpanId:,SpanName:,StartTime:,EndTime:,SpanTags:map[string]string{},}
2015-12-21T01:41:05.191165Z	info	template Kind: kubernetes, &InstanceParam{SourceUid:,SourceIp:,DestinationUid:,DestinationIp:,OriginUid:,OriginIp:,AttributeBindings:map[string]string{},}
2015-12-21T01:41:05.191111Z	info	template Kind: authorization, &InstanceParam{Subject:nil,Action:nil,}
2015-12-21T01:41:05.191136Z	info	template Kind: logentry, &InstanceParam{Variables:map[string]string{},Timestamp:,Severity:,MonitoredResourceType:,MonitoredResourceDimensions:map[string]string{},}
2015-12-21T01:41:05.191260Z	info	template Kind: metric, &InstanceParam{Value:,Dimensions:map[string]string{},MonitoredResourceType:,MonitoredResourceDimensions:map[string]string{},}
2015-12-21T01:41:05.191208Z	info	template Kind: quota, &InstanceParam{Dimensions:map[string]string{},}
2015-12-21T01:41:05.191224Z	info	template Kind: reportnothing, &InstanceParam{}
2015-12-21T01:41:05.191258Z	info	template Kind: servicecontrolreport, &InstanceParam{ApiVersion:,ApiOperation:,ApiProtocol:,ApiService:,ApiKey:,RequestTime:,RequestMethod:,RequestPath:,RequestBytes:,ResponseTime:,ResponseCode:,ResponseBytes:,ResponseLatency:,}
2015-12-21T01:41:05.191370Z	info	template Kind: apikey, &InstanceParam{Api:,ApiVersion:,ApiOperation:,ApiKey:,Timestamp:,}
2015-12-21T01:41:05.191399Z	info	template Kind: checknothing, &InstanceParam{}
2015-12-21T01:41:05.191313Z	info	template Kind: listentry, &InstanceParam{Value:,}
2015-12-21T01:41:05.191341Z	info	adapter Kind: opa, &Params{Policy:[],CheckMethod:,FailClose:false,}
2015-12-21T01:41:05.191463Z	info	template Kind: rule
2015-12-21T01:41:05.191482Z	info	template Kind: attributemanifest
2015-12-21T01:41:05.213480Z	info	Published snapshot[1] with 1 rules, 1 handlers, previously 0 rules
2015-12-21T01:41:05.213411Z	info	Config controller has started with 4 config elements
2015-12-21T01:41:05.220535Z	error	Check returned with error: PERMISSION_DENIED (opaHandler.opa.istio-system:opa: request was rejected)
2015-12-21T01:41:05.221921Z	error	Check returned with error: PERMISSION_DENIED (opaHandler.opa.istio-system:opa: request was rejected)
2015-12-21T01:41:05.224461Z	error	Check returned with error: PERMISSION_DENIED (opaHandler.opa.istio-system:opa: request was rejected)
--- FAIL: TestServer (0.04s)
	opa_integration_test.go:282: Not important API: Expected: 0, Got: 7
	opa_integration_test.go:282: Self permission: Expected: 0, Got: 7
--- FAIL: TestValidateError (0.00s)
	opa_test.go:161: Invalid policies: expected: Policy: 1 error occurred: opa_policy.0:3: rego_parse_error: no match found, unexpected '+', |||received: Policy: 1 error occurred: opa_policy.0:2: rego_parse_error: no match found, expected: "#", "-", ".", "0", "[", "\"", "`", "default", "false", "import", "not", "null", "package", "set(", "true", "{", [ \t\r\n], [ \t], [1-9], [A-Za-z_] or EOF
--- FAIL: TestMultiplePolicy (0.03s)
	opa_test.go:393: Got status 7, expecting OK
	opa_test.go:393: Got status 7, expecting OK
	opa_test.go:393: Got status 7, expecting OK
	opa_test.go:393: Got status 7, expecting OK
	opa_test.go:393: Got status 7, expecting OK
	opa_test.go:393: Got status 7, expecting OK
	opa_test.go:393: Got status 7, expecting OK
FAIL
FAIL	istio.io/istio/mixer/adapter/opa	0.366s

--- a/istioctl/pkg/install/pre-check.go
+++ b/istioctl/pkg/install/pre-check.go
@@ -83,9 +83,8 @@ func installPreCheck(istioNamespaceFlag string, restClientGetter genericclioptio
 		errs = multierror.Append(errs, fmt.Errorf("failed to initialize the Kubernetes client: %v", err))
 		fmt.Fprintf(writer, "Failed to initialize the Kubernetes client: %v.\n", err)
 		return errs
-	} else {
-		fmt.Fprintf(writer, "Can initialize the Kubernetes client.\n")
 	}
+	fmt.Fprintf(writer, "Can initialize the Kubernetes client.\n")
 	v, err := c.serverVersion()
 	if err != nil {
 		errs = multierror.Append(errs, fmt.Errorf("failed to query the Kubernetes API Server: %v", err))

--- a/istioctl/pkg/install/pre-check.go
+++ b/istioctl/pkg/install/pre-check.go
@@ -82,6 +82,7 @@ func installPreCheck(istioNamespaceFlag string, restClientGetter genericclioptio
 	if err != nil {
 		errs = multierror.Append(errs, fmt.Errorf("failed to initialize the Kubernetes client: %v", err))
 		fmt.Fprintf(writer, "Failed to initialize the Kubernetes client: %v.\n", err)
+		return errs
 	} else {
 		fmt.Fprintf(writer, "Can initialize the Kubernetes client.\n")
 	}

